extensions/mssql/src/contracts.ts

```diff
@@ -36,19 +36,48 @@ export interface RequestSecurityTokenParams {
 	authority: string;
 	provider: string;
 	resource: string;
+	accountId: string;
 	scope: string;
 }
 
 export interface RequestSecurityTokenResponse {
 	accountKey: string;
 	token: string;
+	expiresOn: Number;
 }
 
 export namespace SecurityTokenRequest {
 	export const type = new RequestType<RequestSecurityTokenParams, RequestSecurityTokenResponse, void, void>('account/securityTokenRequest');
 }
 // ------------------------------- </ Security Token Request > ------------------------------------------
 
+// ------------------------------- < Refresh Token Request > ---------------------------------
+
+export interface RefreshTokenNotificationParams {
+	authority: string;
+	provider: string;
+	resource: string;
+	accountId: string;
+	uri: string;
```

Comment:
I'm not sure I like tying this to a specific URI. The tokens are more generic than that - a single token is scoped to account/authority/provider/resource but can be used across multiple different sources (like multiple queries). 

If possible it would be great to have this notification be just for the token itself and shared across URIs

Reply:
This is because on the STS side we use the URI to find the connection